#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/RectangleArena.proto"
IMPORTABLE EXTERNPROTO "../protos/maze_wall.proto"

WorldInfo {
  contactProperties [
    ContactProperties {
      softCFM 0.0001
    }
  ]
}
Viewpoint {
  orientation -0.29248493065817743 0.15543148697706657 0.9435537177044978 2.250307395927664
  position 7.421046746590603 -4.520099152429031 5.83518106678849
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  translation 3.5 3.5 0
  floorSize 7.5 7.5
  floorAppearance PBRAppearance {
    baseColor 0 0 0
    roughness 1
    metalness 0
  }
}
Robot {
  translation 0.4 0 0
  children [
    DistanceSensor {
      translation 0 0.06 0.03
      rotation 0 0 1 1.57
      children [
        DEF distanceBoundBox Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.02 0.02 0.02
          }
        }
      ]
      name "sonar"
      boundingObject USE distanceBoundBox
      physics Physics {
      }
      lookupTable [
        0 0 0
        0.3 3000 0
      ]
    }
    Solid {
      children [
        DEF bdy Group {
          children [
            Pose {
              translation 0 0.05 0.15
              rotation 1 0 0 1.57
              children [
                Shape {
                  appearance PBRAppearance {
                    baseColor 0 1 0
                    roughness 1
                    metalness 0
                  }
                  geometry Cylinder {
                    height 0.005
                    radius 0.01
                  }
                }
              ]
            }
            Pose {
              translation 0 0 0.1
              children [
                Shape {
                  appearance PBRAppearance {
                    roughness 1
                    metalness 0
                  }
                  geometry Cylinder {
                    height 0.1
                    radius 0.05
                  }
                }
              ]
            }
            Pose {
              translation 0 0 0.035
              children [
                Shape {
                  appearance PBRAppearance {
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 0.1 0.1 0.05
                  }
                }
              ]
            }
          ]
        }
      ]
      name "solid(4)"
      boundingObject USE bdy
      physics Physics {
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.055 -0.04 0.02
      }
      device [
        RotationalMotor {
          name "bwdL"
        }
      ]
      endPoint Solid {
        translation -0.055 -0.04 0.02
        rotation 0 1 0 1.57
        children [
          DEF wheelShape Shape {
            appearance PBRAppearance {
              baseColor 0 1 1
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.01
              radius 0.02
            }
          }
        ]
        name "bwdL_solid"
        boundingObject USE wheelShape
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.055 -0.04 0.02
      }
      device [
        RotationalMotor {
          name "bwdR"
        }
      ]
      endPoint Solid {
        translation 0.055 -0.04 0.02
        rotation 0 1 0 1.57
        children [
          USE wheelShape
        ]
        name "bwdR_solid"
        boundingObject USE wheelShape
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.055 0.04 0.02
      }
      device [
        RotationalMotor {
          name "fwdL"
        }
      ]
      endPoint Solid {
        translation -0.055 0.04 0.02
        rotation 0 1 0 1.57
        children [
          USE wheelShape
        ]
        name "fwdL_solid"
        boundingObject USE wheelShape
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.055 0.04 0.02
      }
      device [
        RotationalMotor {
          name "fwdR"
        }
      ]
      endPoint Solid {
        translation 0.055 0.04 0.02
        rotation 0 1 0 1.57
        children [
          USE wheelShape
        ]
        name "fwdR_solid"
        boundingObject USE wheelShape
        physics Physics {
        }
      }
    }
    DistanceSensor {
      translation -0.05 -0.03 0.06
      rotation 0 0 1 -3.14
      children [
        DEF distanceBoundBox Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.02 0.02 0.02
          }
        }
      ]
      name "sonarL_bwd"
      boundingObject USE distanceBoundBox
      physics Physics {
      }
      lookupTable [
        0 0 0
        0.5 5000 0
      ]
    }
    DistanceSensor {
      translation 0.05 -0.03 0.06
      children [
        DEF distanceBoundBox Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.02 0.02 0.02
          }
        }
      ]
      name "sonarR_bwd"
      boundingObject USE distanceBoundBox
      physics Physics {
      }
      lookupTable [
        0 0 0
        0.5 5000 0
      ]
    }
    DistanceSensor {
      translation -0.05 0.04 0.06
      rotation 0 0 1 3.14
      children [
        DEF distanceBoundBox Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.02 0.02 0.02
          }
        }
      ]
      name "sonarL_fwd"
      boundingObject USE distanceBoundBox
      physics Physics {
      }
      lookupTable [
        0 0 0
        0.5 5000 0
      ]
    }
    DistanceSensor {
      translation 0.05 0.04 0.06
      children [
        DEF distanceBoundBox Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.02 0.02 0.02
          }
        }
      ]
      name "sonarR_fwd"
      boundingObject USE distanceBoundBox
      physics Physics {
      }
      lookupTable [
        0 0 0
        0.5 5000 0
      ]
    }
    DistanceSensor {
      translation 0 0 0.01
      rotation 0 1 0 1.5700003061004253
      children [
        DEF irBound Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "irsense"
      boundingObject USE irBound
      physics Physics {
      }
      type "infra-red"
    }
  ]
  boundingObject DEF bdy Group {
    children [
      Pose {
        translation 0 0 0.1
        children [
          Shape {
            appearance PBRAppearance {
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.1
              radius 0.05
            }
          }
        ]
      }
      Pose {
        translation 0 0 0.028
        children [
          Shape {
            appearance PBRAppearance {
              roughness 1
              metalness 0
            }
            geometry Box {
              size 0.1 0.1 0.05
            }
          }
        ]
      }
    ]
  }
  physics Physics {
  }
  controller "my_controller"
}
Robot {
  name "robot(1)"
  controller "my_supervisor"
  supervisor TRUE
}
